File: CS/DataModel.Shared/BusinessObjects/Employee.cs
using System.ComponentModel.DataAnnotations;
using DevExpress.Persistent.BaseImpl.EF;

namespace DataModel.Shared.BusinessObjects {
    public class Employee : BaseObject {

        [Required()]
        public virtual String FirstName { get; set; }

        [Required()]
        public virtual String LastName { get; set; }

        public virtual DateTime? Birthday { get; set; }

        [MaxLength(255)]
        public virtual String Email { get; set; }

        public virtual Department Department { get; set; }

        public string FullName {
            get { return $"{FirstName} {LastName}"; }
        }

        public void CopyToContextObject(Employee contextObj, DXApplication1EFCoreDbContext dbContext) {
            contextObj.FirstName = this.FirstName;
            contextObj.LastName = this.LastName;
            contextObj.Birthday = this.Birthday;
            contextObj.Email = this.Email;
            if(this.Department != null)
                contextObj.Department = dbContext.Departments.First(n => n.ID == this.Department.ID);
            else
                contextObj.Department = null;
        }
    }
}

File: VB/DataModel.Shared/BusinessObjects/Employee.vb
Imports System.ComponentModel.DataAnnotations
Imports DevExpress.Persistent.BaseImpl.EF

Namespace DataModel.Shared.BusinessObjects

    Public Class Employee
        Inherits BaseObject

        <Required()>
        Public Overridable Property FirstName As [String]

        <Required()>
        Public Overridable Property LastName As [String]

        Public Overridable Property Birthday As DateTime?

        <MaxLength(255)>
        Public Overridable Property Email As [String]

        Public Overridable Property Department As Department

        Public ReadOnly Property FullName As String
            Get
                Return $"{FirstName} {LastName}"
            End Get
        End Property

        Public Sub CopyToContextObject(ByVal contextObj As Employee, ByVal dbContext As DXApplication1EFCoreDbContext)
            contextObj.FirstName = FirstName
            contextObj.LastName = LastName
            contextObj.Birthday = Birthday
            contextObj.Email = Email
            If Department IsNot Nothing Then
                contextObj.Department = dbContext.Departments.First(Function(n) n.ID = Department.ID)
            Else
                contextObj.Department = Nothing
            End If
        End Sub
    End Class
End Namespace
